complex/ft_msg.c

```diff
@@ -290,6 +309,48 @@ int ft_recv_dgram(void)
 	return -FI_ETIMEDOUT;
 }
 
+int ft_recv_dgram_flood(void)
+{
+	struct timespec s, e;
+	int credits, ret;
+	int64_t poll_time = 0;
+	int expected = ft.xfer_iter;
+	int got	     = 0;
+	do {
+		if (ft_rx.credits > (ft_rx.max_credits >> 1)) {
+			ret = ft_post_recv_bufs();
+			if (ret)
+				return ret;
+		}
+
+		credits = ft_rx.credits;
+
+		ret = ft_comp_rx();
+		if (credits != ft_rx.credits) {
+			int num_creds = ft_rx.credits - credits;
+			poll_time=0;
+			got+=num_creds;
+		}
+		if (ret)
+			return ret;
+
+		if(got >= expected) {
+			return 0;
```

Comment:
We might have to bite the bullet and use multiple receive buffers…it will be required if we want to add data integrity checks to the buckets.


> On Apr 8, 2015, at 3:22 PM, Sean Hefty <notifications@github.com> wrote:
> 
> In complex/ft_msg.c <https://github.com/ofiwg/fabtests/pull/221#discussion_r28020011>:
> 
> > +				return ret;
> > +		}
> > +
> > +		credits = ft_rx.credits;
> > +
> > +		ret = ft_comp_rx();
> > +		if (credits != ft_rx.credits) {
> > +			int num_creds = ft_rx.credits - credits;
> > +			poll_time=0;
> > +			got+=num_creds;
> > +		}
> > +		if (ret)
> > +			return ret;
> > +
> > +		if(got >= expected) {
> > +			return 0;
> I was just looking at how to deal with this:
> 1) send immediate data.
> This seems like the best option -- if only it would work on iwarp. :/
> 2) use multiple receive buffers
> This looks like it requires a fair amount of rework. :/
> 3) use another endpoint
> The receiver would need to inspect which EP each message arrived on, or look at multiple CQs.
> —
> Reply to this email directly or view it on GitHub <https://github.com/ofiwg/fabtests/pull/221/files#r28020011>.
>